util/fipstools/acvp/modulewrapper/modulewrapper.cc

```diff
@@ -996,6 +1030,31 @@ static bool HashMCT(const Span<const uint8_t> args[],
       {Span<const uint8_t>(buf + 2 * DigestLength, DigestLength)});
 }
 
+template <size_t DigestLength, const EVP_MD *(MDFunc)()>
+static bool HashMCTSha3(const Span<const uint8_t> args[],
+                    ReplyCallback write_reply) {
+  if (args[0].size() != DigestLength) {
+    return false;
+  }
+  const EVP_MD *evp_md = MDFunc();
+  unsigned int md_out_size = DigestLength;
+
+  unsigned char md[1001][DigestLength];
+  unsigned char msg[1001][DigestLength];
+
+  memcpy(md[0], args[0].data(), DigestLength);
+
+  EVP_MD_unstable_sha3_enable(true);
+  for (size_t i = 1; i <= 1000; i++) {
+    memcpy(msg[i], md[i-1], DigestLength);
+    EVP_Digest(msg[i], sizeof(msg[i]), md[i], &md_out_size, evp_md, NULL);
+  }
+  EVP_MD_unstable_sha3_enable(false);
+
+  return write_reply(
+      {Span<const uint8_t>(md[1000])});
```

Comment:
Should this be 1001? Or md be only 1000 long on line 1042?

Reply:
1001 will be out of index because of the way C creates arrays when they're initialized with 1001 (they create an array with indices 0-1000).